test/main.F90

```diff
@@ -0,0 +1,103 @@
+
+  if (me==1) print *, new_line(''), "_________ In total, ",passes," of ",tests, " tests pass. _________"
+  call prif_sync_all
+  if (passes /= tests) error stop
+
+contains
+
+  subroutine stop_and_print_usage_info_if_help_requested
+    type(command_line_t) command_line
+
+    character(len=*), parameter :: usage = &
+      new_line('') // new_line('') // &
+      'Usage: fpm test -- [--help] | [--contains <substring>]' // &
+      new_line('') // new_line('') // &
+      'where square brackets ([]) denote optional arguments, a pipe (|) separates alternative arguments,' // new_line('') // &
+      'angular brackets (<>) denote a user-provided value, and passing a substring limits execution to' // new_line('') // &
+      'the tests with test subjects or test descriptions containing the user-specified substring.' // new_line('') 
+
+    if (command_line%argument_present([character(len=len("--help"))::"--help","-h"])) stop usage
+
+  end subroutine
+
+  subroutine run_tests_and_report(passes, tests)
+    integer, intent(inout) :: passes, tests
+
+    type(prif_allocate_test_t) prif_allocate_test
+    type(prif_co_broadcast_test_t) prif_co_broadcast_test
+    type(prif_co_max_test_t) prif_co_max_test
+    type(prif_co_min_test_t) prif_co_min_test
+    type(prif_co_reduce_test_t) prif_co_reduce_test
+    type(prif_co_sum_test_t) prif_co_sum_test
+    type(prif_error_stop_test_t) prif_error_stop_test
+    type(prif_image_index_test_t) prif_image_index_test
+    type(prif_init_test_t) prif_init_test
+    type(prif_num_images_test_t) prif_num_images_test
+    type(prif_rma_test_t) prif_rma_test
+    type(prif_stop_test_t) prif_stop_test
+    type(prif_teams_test_t) prif_teams_test
+    type(prif_this_image_test_t) prif_this_image_test
+
+    call prif_init_test%report(passes, tests)
+    call prif_allocate_test%report(passes, tests)
+    call prif_num_images_test%report(passes, tests)
+    call prif_this_image_test%report(passes, tests)
+    call prif_co_broadcast_test%report(passes, tests)
+    call prif_teams_test%report(passes, tests)
+#ifdef __flang__
+    print *
+    print *,"!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
+    print *
+    print *,"LLVM Flang detected. Skipping tests that crash:"
+    print *,"  - prif_co_max_test"
+    print *,"  - prif_co_min_test"
+    print *,"  - prif_co_reduce_test"
+    print *,"  - prif_co_sum_test"
+    print *,"  - prif_error_stop_test"
+    print *,"  - prif_image_index_test"
+    print *,"  - prif_stop_test"
+    print *
+    print *,"!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
+#else
+    call prif_co_max_test%report(passes, tests)
+    call prif_co_min_test%report(passes, tests)
+    call prif_co_reduce_test%report(passes, tests)
+    call prif_co_sum_test%report(passes, tests)
+    call prif_error_stop_test%report(passes, tests)
+    call prif_image_index_test%report(passes, tests)
+    call prif_rma_test%report(passes, tests)
+    call prif_stop_test%report(passes, tests)
+#endif
```

Comment:
This evokes a somewhat philosophical question about how to proceed in the face of known failures.

Reply:
@everythingfunctional that's a solid point.  At this point, I'd like to understand and address the issues even if the only outcome is that Caffeine and/or Julienne improve whether or not we decide for Caffeine to use Julienne.  So far, these issues have helped to identify problems in both packages that I'm working on addressing.  Most of the above failures relate the handling of `character` stop codes in error termination (see issue #77).  I'm making good progress with addressing the above failures and hope to have most or all fixed by the time of tomorrow's call.